packages/sdk/contractkit/src/wrappers/GrandaMento.test.ts

```diff
@@ -0,0 +1,158 @@
+import { Address } from '@celo/base/lib/address'
+import { NetworkConfig, testWithGanache, timeTravel } from '@celo/dev-utils/lib/ganache-test'
+import BigNumber from 'bignumber.js'
+import Web3 from 'web3'
+import { StableToken as StableTokenName } from '../base'
+import { StableToken } from '../celo-tokens'
+import { newKitFromWeb3 } from '../kit'
+import { GoldTokenWrapper } from './GoldTokenWrapper'
+import { ExchangeProposalState, GrandaMentoWrapper } from './GrandaMento'
+import { StableTokenWrapper } from './StableTokenWrapper'
+
+const expConfig = NetworkConfig.grandaMento
+
+testWithGanache('GrandaMento Wrapper', (web3: Web3) => {
+  const kit = newKitFromWeb3(web3)
+  let accounts: Address[] = []
+  let grandaMento: GrandaMentoWrapper
+  let celoToken: GoldTokenWrapper
+  let stableToken: StableTokenWrapper
+  const newLimitMin = new BigNumber('1000')
+  const newLimitMax = new BigNumber('1000000000000')
+  let sellAmount: BigNumber
+
+  beforeAll(async () => {
+    accounts = await web3.eth.getAccounts()
+    kit.defaultAccount = accounts[0]
+    grandaMento = await kit.contracts.getGrandaMento()
+
+    stableToken = await kit.contracts.getStableToken(StableToken.cUSD)
+    celoToken = await kit.contracts.getGoldToken()
+  })
+
+  const increaseLimits = async () => {
+    await (
+      await grandaMento.setStableTokenExchangeLimits(
+        'StableToken',
+        newLimitMin.toString(),
+        newLimitMax.toString()
+      )
+    ).sendAndWaitForReceipt()
+  }
+
+  describe('No limits sets', () => {
+    it('gets the proposals', async () => {
+      const activeProposals = await grandaMento.getActiveProposalIds()
+      expect(activeProposals).toEqual([])
+    })
+
+    it('fetches empty limits', async () => {
+      const limits = await grandaMento.stableTokenExchangeLimits(StableTokenName.cUSD)
+      expect(limits.minExchangeAmount).toEqBigNumber(new BigNumber(0))
+      expect(limits.maxExchangeAmount).toEqBigNumber(new BigNumber(0))
+    })
+  })
+
+  it("fetchs a proposal it doesn't exist", async () => {
+    const throwFunc = async () => {
+      await grandaMento.getExchangeProposal(0)
+    }
+    await expect(throwFunc).rejects.toThrow("Proposal doesn't exist")
```

Comment:
Why not just `await expect(grandaMento.getExchangeProposal(0)).rejects.toThrow("Proposal doesn't exist") `